HARK/ConsumptionSaving/ConsPortfolioModel.py

```diff
@@ -429,6 +431,39 @@ def solve_one_period_ConsPortfolio(
     RiskyMax = np.max(Risky_next)
     RiskyMin = np.min(Risky_next)
 
+    # Perform an alternate calculation of the absolute patience factor when
+    # returns are risky. This uses the Merton-Samuelson limiting risky share,
+    # which is what's relevant as mNrm goes to infinity.
+    def calc_Radj(R):
+        Rport = ShareLimit * R + (1.0 - ShareLimit) * Rfree
+        return Rport ** (1.0 - CRRA)
+
+    R_adj = expected(calc_Radj, RiskyDstn)[0]
+    PatFac = (DiscFacEff * R_adj) ** (1.0 / CRRA)
+    MPCminNow = 1.0 / (1.0 + PatFac / solution_next.MPCmin)
+
+    # Also perform an alternate calculation for human wealth under risky returns
+    def calc_hNrm(S):
+        Risky = S["Risky"]
+        PermShk = S["PermShk"]
+        TranShk = S["TranShk"]
+        G = PermGroFac * PermShk
+        Rport = ShareLimit * Risky + (1.0 - ShareLimit) * Rfree
+        hNrm = (G / Rport**CRRA) * (TranShk + solution_next.hNrm)
+        return hNrm
+
+    # This correctly accounts for risky returns and risk aversion
+    hNrmNow = expected(calc_hNrm, ShockDstn) / R_adj
+
+    # This basic equation works if there's no correlation among shocks
+    # hNrmNow = (PermGroFac/Rfree)*(1 + solution_next.hNrm)
+
+    # The above attempts to pin down the limiting consumption function for this
+    # model, however it is not clear why it creates bugs, so for now we allow
+    # for a linear extrapolation beyond the last asset point
+    cFuncLimitIntercept = MPCminNow * hNrmNow
+    cFuncLimitSlope = MPCminNow
```

Comment:
That's the question I raised in my last post on the PR thread. The math is correct, and these are the correct terms of the limiting linear cFunc. However, if `aXtraMax` is set too high (say, around 1000 or 10000) then the numerically constructed cFunc ends up with its MPC not converging to the proper MPCmin, which causes issues in the extrapolation. This is due to extreme sensitivity to the accuracy of the ShareFunc.

In the basic solver, I added a "refinement" step to the RiskyShare search that makes the ShareFunc orders of magnitude more accurate. This prevents the cFunc from violating its limit out to about 10B or more; beyond that point, the amount by which it would violate the limiting function is (proportionally) around machine precision, so it can't have any effect.

Putting in the "refinement" step into the full blown portfolio solver is somewhat harder, and I'm a little unenthused about doing it right now. Thus my three options:

1) Don't use the limiting cFunc for now, to keep things safe. Return to it later to finish the job.

2) Use the limiting cFunc in this solver (one line change), and be aware that it will break if `aXtraMax` is set to an unusually high level. Return to finish the job later.

3) Don't merge the PR until it's fully proper.